app/src/test/java/protect/card_locker/contentprovider/AboutActivityUITests.kt

```diff
@@ -0,0 +1,347 @@
+            on { getString(eq(R.string.translate_platform)) } doReturn "Weblate"
+            on { getString(eq(R.string.app_license)) } doReturn "MIT"
+            on { getString(eq(R.string.on_github)) } doReturn "GitHub"
+            on { getString(eq(R.string.and_data_usage)) } doReturn "and data usage"
+            on { getString(eq(R.string.ok)) } doReturn "OK"
+            on { getString(eq(R.string.view_online)) } doReturn "View online"
+            on { getPackageName() } doReturn "com.example.catima"
+            on { getPackageName() } doReturn "protect.card_locker.contentprovider"
+            on { packageManager } doReturn mockPackageManager
+            on {resources} doReturn mockResources
+            }
+
+        activity.content = MockAboutContent(mockContext)
+        mockOpenWebLinkHandler = mock()
+        mockBackPressedDispatcher = mock()
+    }
+
+    class MockAboutContent(context: Context) : AboutContent(context) {
+        override fun getContributorsHtml(): String = "<br/>Test Contributor"
+        override fun getHistoryHtml(): String = "<p>Version 1.0</p>"
+        override fun getLicenseHtml(): String = "<p>MIT License</p>"
+        override fun getPrivacyHtml(): String = "<p>Privacy Policy</p>"
+        override fun getThirdPartyLibrariesHtml(): String = "<br/>Test Library"
+        override fun getUsedThirdPartyAssetsHtml(): String = "<br/>Test Asset"
+        override fun getCurrentYear(): Int = 2025
+    }
+    // Mock OpenWebLinkHandler to verify URL opening
+    class MockOpenWebLinkHandler : OpenWebLinkHandler() {
+        override fun openBrowser(activity: Activity?, url: String?) {
+            val intent = Intent(Intent.ACTION_VIEW, Uri.parse(url))
+            activity?.startActivity(intent)
+        }
+    }
+
+    @Test
+    fun testTopAppBarTitleAndBackButton() {
+        composeTestRule.setContent {
+           CatimaTopAppBar(
+                title = "About Catima",
+                onBackPressedDispatcher = mockBackPressedDispatcher
+            )
+        }
+
+        // Verify title
+        composeTestRule.onNodeWithText("About Catima").assertIsDisplayed()
+
+        // Verify back button and click
+        composeTestRule.onNodeWithContentDescription("Back")
+            .assertIsDisplayed()
+            .performClick()
+
+        // Verify back button triggers onBackPressed
+        verify(mockBackPressedDispatcher).onBackPressed()
+    }
+
+    @Test
+    fun testTopAppBarNoBackButtonWhenDispatcherNull() {
+        composeTestRule.setContent {
+            CatimaTopAppBar(
+                title = "About Catima",
+                onBackPressedDispatcher = null
+            )
+        }
+
+        // Verify title
+        composeTestRule.onNodeWithText("About Catima").assertIsDisplayed()
+
+        // Verify back button is not displayed
+        composeTestRule.onNodeWithContentDescription("Back").assertDoesNotExist()
+    }
```

Comment:
These are more generic UI elements, not limited to the AboutActivity (I want to use these everywhere). Maybe these should be in a separate test file?